sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/UserIdContext.java

```diff
@@ -153,21 +148,35 @@ public synchronized String getUserId() {
      *
      * @param userId user identifier.
      */
-    public synchronized void setUserId(String userId) {
-        mUserId = userId;
+    public void setUserId(String userId) {
+        if (checkEqualsUserId(userId)) {
+            return;
+        }
 
-        // TODO if mUserId != userId
-        // TODO move out from synchronized
         /* Call listeners so that they can react on new token. */
         for (Listener listener : mListeners) {
             listener.onNewUserId(mUserId);
         }
     }
 
+    /**
+     * Check user identifier.
+     *
+     * @param userId user identifier.
+     * @return true if equals.
+     */
+    private synchronized boolean checkEqualsUserId(String userId) {
```

Comment:
Reconsider the name

Reply:
`equals`, too? I don't think we need this one